### DISCUSSION TITLE
Cannot access api before initialisation when using with react router loader

### DISCUSSION THREAD

**Discussion by AsuraKev** [reduxjs/redux-toolkit#3825] [Q&A]
Hi there, I am trying to dispatch an RTK query fetch inside the loader which I exported and used inside the routing config like below. but I am getting this initialisation error... pretty sure redux has already been setup and running ....

Need help! :) thanks

![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/f3d05203-315b-492b-8cc8-9d1378cad2dc)

![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/032c9174-3afd-496c-88ee-282af3034d45)

![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/5cd832e3-68f7-43d4-84e6-308426c911d1)
![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/37f10622-4bd5-473f-a568-fb6f5056d03c)

**Comment by markerikson** [comment]
Seems like there might be a circular import issue of some kind, but hard to tell without knowing your codebase.

**Reply by AsuraKev** [reply]
hmm I am thinking thats the case too, but I still dont get why its a circular reference. By commenting out below code, the error is gone :/

![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/1ab45f89-a35f-4999-8f8d-00c7ba04573f)

**Comment by AsuraKev** [comment]
turns out the order of import screwed up. I tried to import the RTK query API before the importing the redux store... 
![image](https://github.com/reduxjs/redux-toolkit/assets/52765572/75eedbcb-3212-4b2d-aa0f-74430df63ebb)
